### PR TITLE
feat(route): add 色花堂作者文章

### LINKED ISSUE #11698: 请求添加色花堂作者文章
### 类型

社交媒体

### 网站地址

https://sehuatang.org

### 网站描述

有许多用户的成人论坛

### 需要生成什么内容？

生成某个作者的文章，目前只有相关的分区文章。

### 额外描述

如：https://sehuatang.org/home.php?mod=space&uid=418520&do=thread&view=me&from=space

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by JamYiz**
## 该 PR 相关 Issue / Involved Issue

Close #11698

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/sehuatang/bt/gcyc
/sehuatang/picture/gcyc
/sehuatang/gcyc/684
/sehuatang/gcyc
/sehuatang/user/411096
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x]  新的路由 New Route
  - [x]  跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x]  文档说明 Documentation
  - [x]  中文文档 CN
  - [ ] 英文文档 EN
- [x]  全文获取 fulltext
  - [x]  使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x]  [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x]  可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
修改了色花堂原本的路由地址，新增了作者文章的规则。

### CODE REVIEW

File: docs/multimedia.md [hunk 1]
```diff
@@ -1619,7 +1619,7 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
 
 ### 分区帖子
 
-<Route author="qiwihui junfengP nczitzk" example="/sehuatang/36/368" path="/sehuatang/:subforumid?/:type?" supportBT="1" :paramsDesc="['版块 id 或板块名称（见下表）, 为空默认高清中文字幕', '类型 id, 可在分区类型过滤后的 URL 中找到']">
+<Route author="qiwihui junfengP nczitzk" example="/sehuatang/subforum/36/368" path="/sehuatang/subforum/:subforumid?/:type?" supportBT="1" :paramsDesc="['版块 id 或板块名称（见下表）, 为空默认高清中文字幕', '类型 id, 可在分区类型过滤后的 URL 中找到']">
```
Comment (TonyRL) [inline docs/multimedia.md:1622]: Completely unnecessary changes.

File: lib/v2/sehuatang/router.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module.exports = function (router) {
-    router.get('/bt/:subforumid?', require('./index'));
-    router.get('/picture/:subforumid', require('./index'));
-    router.get('/:subforumid?/:type?', require('./index'));
-    router.get('/:subforumid?', require('./index'));
-    router.get('', require('./index'));
+    router.get('/subforum/bt/:subforumid?', require('./index'));
+    router.get('/subforum/picture/:subforumid', require('./index'));
+    router.get('/subforum/:subforumid?/:type?', require('./index'));
+    router.get('/subforum/:subforumid?', require('./index'));
```
Comment (TonyRL) [inline lib/v2/sehuatang/router.js:5]: Completely unnecessary changes.

File: docs/multimedia.md [hunk 3]
```diff
@@ -1635,6 +1635,10 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
 
 </Route>
 
+### 作者文章
+
+<Route author="JamYiz" example="/sehuatang/user/411096"path="/sehuatang/user/:uid":paramsDesc="'uid, 必选 - 用户 uid, 可在用户主页 URL 中找到'"/>
```
Comment (TonyRL) [inline docs/multimedia.md:1640]: This route depends on extra config. You need to set `selfhost` attribute.

File: lib/v2/sehuatang/maintainer.js [hunk 4]
```diff
@@ -4,4 +4,5 @@ module.exports = {
     '/:subforumid?/:type?': ['qiwihui', 'junfengP', 'nczitzk'],
     '/:subforumid?': ['qiwihui', 'junfengP', 'nczitzk'],
     '/': ['qiwihui', 'junfengP', 'nczitzk'],
+    '/user/:uid': ['JamYiz']
```
Comment (TonyRL) [inline lib/v2/sehuatang/maintainer.js:7]: Please add new entries in alphabetical order.

**Review by TonyRL — COMMENTED** [review]
Please fix those lint issues before requesting a review.

File: docs/multimedia.md [hunk 5]
```diff
@@ -1635,6 +1635,10 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
 
 </Route>
 
+### 作者文章
+
+\<Route author="JamYiz" example="/sehuatang/user/411096"path="/sehuatang/user/:uid":paramsDesc="\['uid, 必选 - 用户 uid, 可在用户主页 URL 中找到 ']"selfhost="1"/>
```
Comment (TonyRL) [inline docs/multimedia.md:1640]:
```suggestion
<Route author="JamYiz" example="/sehuatang/user/411096" path="/sehuatang/user/:uid" :paramsDesc="['用户 uid, 可在用户主页 URL 中找到']" selfhost="1"/>
```
- Every attribute needs to be separated by a space.
- No need to mention the path parameters and the necessity again.
- Since you've set `selfhost` attribute, you need to update the docs on what envs need to be set in `install/README.md`.

docs: https://docs.rsshub.app/joinus/new-rss/add-docs.html

File: lib/v2/sehuatang/maintainer.js [hunk 6]
```diff
@@ -3,5 +3,6 @@ module.exports = {
     '/picture/:subforumid': ['qiwihui', 'junfengP', 'nczitzk'],
     '/:subforumid?/:type?': ['qiwihui', 'junfengP', 'nczitzk'],
     '/:subforumid?': ['qiwihui', 'junfengP', 'nczitzk'],
+    '/user/:uid': ['JamYiz'],
```
Comment (TonyRL) [inline lib/v2/sehuatang/maintainer.js:6]:
```suggestion
    '/picture/:subforumid': ['qiwihui', 'junfengP', 'nczitzk'],
    '/user/:uid': ['JamYiz'],
    '/:subforumid?/:type?': ['qiwihui', 'junfengP', 'nczitzk'],
    '/:subforumid?': ['qiwihui', 'junfengP', 'nczitzk'],
```

File: lib/v2/sehuatang/router.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 module.exports = function (router) {
+    router.get('/user/:uid', require('./user'));
     router.get('/bt/:subforumid?', require('./index'));
     router.get('/picture/:subforumid', require('./index'));
```
Comment (TonyRL) [inline lib/v2/sehuatang/router.js:4]:
```suggestion
    router.get('/bt/:subforumid?', require('./index'));
    router.get('/picture/:subforumid', require('./index'));
    router.get('/user/:uid', require('./user'));
```
Please add new entries in alphabetical order.
docs: https://docs.rsshub.app/joinus/script-standard.html

File: lib/v2/sehuatang/user.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+const cheerio = require('cheerio'); // 可以使用类似 jQuery 的 API HTML 解析器
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const config = require('@/config').value;
+
+const baseUrl = 'https://sehuatang.org/';
+
+module.exports = async (ctx) => {
+    // 从Url参数中获取uid
+    const uid = ctx.params.uid ?? '186424';
```
Comment (TonyRL) [inline lib/v2/sehuatang/user.js:12]: If you set a default value of a variable, please do mention it in the docs.
docs: https://docs.rsshub.app/joinus/new-rss/add-docs.html
https://docs.rsshub.app/joinus/new-rss/before-start.html

File: lib/v2/sehuatang/user.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+const got = require('@/utils/got'); // 自订的 got
+const cheerio = require('cheerio'); // 可以使用类似 jQuery 的 API HTML 解析器
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const config = require('@/config').value;
+
+const baseUrl = 'https://sehuatang.org/';
+
+module.exports = async (ctx) => {
+    // 从Url参数中获取uid
```
Comment (TonyRL) [inline lib/v2/sehuatang/user.js:11]: https://github.com/DIYgod/RSSHub/blob/9bdba060c7962d734745c8d6b9264da9a532b51d/lib/routes/disqus/posts.js#L5-L7
Similarly, you can throw an error if the cookie is not set.

File: lib/v2/sehuatang/user.js (hunk 8 above)
Comment (JamYiz) [reply lib/v2/sehuatang/user.js]: Ok, I've removed the default value, it's an unnecessary value.

File: lib/v2/sehuatang/user.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+const cheerio = require('cheerio'); // 可以使用类似 jQuery 的 API HTML 解析器
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const config = require('@/config').value;
+
+const baseUrl = 'https://sehuatang.org/';
+
+module.exports = async (ctx) => {
+    if (!config.sehuatang.cookie) {
+        throw 'Sehuatang  RSS is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
```
Comment (TonyRL) [inline lib/v2/sehuatang/user.js:12]:
```suggestion
        throw 'Sehuatang RSS is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
```
